queue/src/test/java/org/killbill/queue/TestReaper.java

```diff
@@ -0,0 +1,254 @@
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestReaper extends TestSetup {
+
+    private static final Long SEARCH_KEY_1 = 1L;
+    private static final Long SEARCH_KEY_2 = 2L;
+
+    private DBBackedQueue<BusEventModelDao> queue;
+    private PersistentBusSqlDao sqlDao;
+
+    @BeforeClass(groups = "slow")
+    public void beforeClass() throws Exception {
+        super.beforeClass();
+        sqlDao = getDBI().onDemand(PersistentBusSqlDao.class);
+    }
+
+    @BeforeMethod(groups = "slow")
+    public void beforeMethod() throws Exception {
+        super.beforeMethod();
+        final List<BusEventModelDao> ready = sqlDao.getReadyEntries(clock.getUTCNow().toDate(), 100, null, "bus_events");
+        assertEquals(ready.size(), 0);
+    }
+
+    @Test(groups = "slow")
+    public void testReapEntries() {
+        final PersistentBusConfig config = createConfig();
+        queue = new DBBackedQueueWithPolling<BusEventModelDao>(clock, dbi, PersistentBusSqlDao.class, config, "testReapEntries", metricRegistry);
+        final DateTime now = clock.getUTCNow();
+
+        // Insert ready entry (not yet reapable)
+        sqlDao.insertEntry(createEntry(1L, CreatorName.get(), null, now, PersistentQueueEntryLifecycleState.AVAILABLE), config.getTableName());
+        // Insert ready entry from another node (not yet reapable)
+        sqlDao.insertEntry(createEntry(2L, "thatOtherNode", null, now, PersistentQueueEntryLifecycleState.AVAILABLE), config.getTableName());
+        // Insert in-processing entry (not yet reapable)
+        sqlDao.insertEntry(createEntry(3L, CreatorName.get(), CreatorName.get(), now, PersistentQueueEntryLifecycleState.IN_PROCESSING), config.getTableName());
+        // Insert in-processing entry from another node (not yet reapable)
+        sqlDao.insertEntry(createEntry(4L, "thatOtherNode", "thatOtherNode", now, PersistentQueueEntryLifecycleState.IN_PROCESSING), config.getTableName());
+        // Insert in-processing entry (reapable)
+        sqlDao.insertEntry(createEntryForReaping(5L, CreatorName.get(), CreatorName.get(), now, config.getReapThreshold().getMillis(), PersistentQueueEntryLifecycleState.IN_PROCESSING), config.getTableName());
+        // Insert in-processing entry from another node (reapable)
+        sqlDao.insertEntry(createEntryForReaping(6L, "thatOtherNode", "thatOtherNode", now, config.getReapThreshold().getMillis(), PersistentQueueEntryLifecycleState.IN_PROCESSING), config.getTableName());
+
+        // Check ready entries
+        final List<BusEventModelDao> readyEntries = sqlDao.getReadyEntries(now.toDate(), 10, CreatorName.get(), config.getTableName());
+        // One ready entry (STICKY_POLLING mode)
+        assertEquals(readyEntries.size(), 1);
+        assertEquals(readyEntries.get(0).getRecordId(), (Long) 1L);
+
+        // Check ready and in processing entries
```

Comment:
The last query checks for this node only but this one is global across nodes if i understand correctly.

Reply:
Yup